.travis.yml

```diff
@@ -46,9 +46,15 @@ before_script:
 
 # Generate documentation, compile the engine, run tests.
 script: |
-  cargo install rustfmt &&
-  cargo install cargo-update &&
-  cargo install-update rustfmt &&
+  # Check if cargo-update is present
+  if ![cargo install-update --version]; then
+    cargo install cargo-update;
+  fi &&
+  if ![rustfmt --version]; then
+    cargo install rustfmt;
```

Comment:
Should be rustfmt-nightly

Reply:
From the rustfmt readme (emphasis mine)

(These instructions use the Syntex version of Rustfmt. If you want to use the nightly version replace install rustfmt with install rustfmt-nightly, **however you must then only run this with the nightly toolchain**)

Meaning this would break our stable and beta builds.

Reply:
We could run it as a separate [build stage](https://docs.travis-ci.com/user/build-stages/). It'd have to run after tests though.